WPMedia/arc-themes-components
# TMEDIA-773 - Carousel Component without Library

## Ticket

- [TMEDIA-773](https://arcpublishing.atlassian.net/browse/TMEDIA-773)

## Description

Implementation of a carousel component without using a third party carousel library.

Item of note that may appear different to that of the acceptance criteria of the ticket is the implementation of the buttons - I changed the approach to not use `<Carousel.Button>` sub components to reduce complexity of managing how the events would have to be passed to each button.

## Test Steps

1. Checkout branch - `git checkout TMEDIA-773-carousel`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the button storybook documentation

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
